commit/chainfee/validate_observation.go

```diff
@@ -39,35 +37,39 @@ func (p *processor) ValidateObservation(
 	}
 
 	for _, feeComponent := range obs.FeeComponents {
-		err := validateBigInt(feeComponent.ExecutionFee, "execution fee")
+		err := validateBigInt(feeComponent.ExecutionFee, "execution fee", false)
 		if err != nil {
 			return err
 		}
 
-		err = validateBigInt(feeComponent.DataAvailabilityFee, "data availability fee")
+		err = validateBigInt(feeComponent.DataAvailabilityFee, "data availability fee", true)
 		if err != nil {
 			return err
 		}
 
 	}
 
 	for _, token := range obs.NativeTokenPrices {
-		err := validateBigInt(token.Int, "native token price")
+		err := validateBigInt(token.Int, "native token price", false)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func validateBigInt(b *big.Int, name string) error {
+func validateBigInt(b *big.Int, name string, allowZero bool) error {
 	zero := big.NewInt(0)
 	if b == nil {
 		return fmt.Errorf("nil %s", name)
 	}
-	if b.Cmp(zero) < 0 || b.Cmp(zero) == 0 {
+	if b.Cmp(zero) < 0 {
 		return fmt.Errorf("%s must be positive", name)
 	}
+	if !allowZero && b.Cmp(zero) == 0 {
+		return fmt.Errorf("%s must be non zero", name)
+	}
+
 	return nil
```

Comment:
ValidateBigInt - returns an error if number is negative or if number is zero and we don't allow zero


I think instead of
```
if err = validate(v, "some name", true); err != nil { return err}
```

i recommend
```
if v.Cmp(zero) <= 0 { return fmt.Errorf("some name...") } // not positive
if v.Cmp(zero) < 0 { return fmt.Errorf("some name...") } // is negative
```

IMO the extra function makes things more complex to understand.